src/python/WMComponent/DBS3Buffer/DBSBufferUtil.py

```diff
@@ -202,3 +204,38 @@ def getCompletedWorkflows(self):
         result = wfCompletedDAO.execute(transaction = False)
 
         return result
+    
+    def getPhEDExDBSStatusForCompletedWorkflows(self, summary=False):
+        
+        """
+        _gettPhEDxDBSStatuForCompletedWorkflows_
+
+        """
+        wfCompletedStatusDAO = self.daoFactory(classname = "CheckStatusForCompletedWorkflows")
+        result = wfCompletedStatusDAO.execute(transaction = False)
+        if summary:
+            result = self.summaryPhEDExDBSStatus(result)
+            
+        return result
+
+    def summaryPhEDExDBSStatus(self, data):
+        """
+        """
+        summary = defaultdict(dict)
+        for workflow, value in data.iteritems():
+            # only getting completed workflows
+            summary[workflow]["Completed"] = True
+            
+            if value["NotInPhEDEx"] == 0 and value["InPhEDEx"] > 0:
```

Comment:
I think it needs be handled by EmptyTask below. Which means everything is failed.